### PR TITLE
Edge secure endpoint

### PR DESCRIPTION
**Pull request by amosmintzcyberark**
### Desired Outcome

*Please describe the desired outcome for this PR.  Said another way, what was
the original request that resulted in these code changes?  Feel free to copy
this information from the connected issue.*

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue ID: [insert issue ID]

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: app/controllers/edge_controller.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class EdgeController < RestController
+
+  def all_secrets
+    allowed_params = %i[account limit offset]
+    options = params.permit(*allowed_params)
+      .slice(*allowed_params).to_h.symbolize_keys
```
Comment (yoavgeva) [inline app/controllers/edge_controller.rb:8]: you sure you need slice? it's just do a duplicate of list

Comment (yoavgeva) [inline app/controllers/edge_controller.rb:8]: you sure you need slice? it's just do a duplicate of list

File: app/controllers/edge_controller.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        variableToReturn[:value] = variable.last_secret.value
+      end
+      results  << variableToReturn
+    end
+    render(json: results)
+  end
+
+  def all_hosts
+    allowed_params = %i[account limit offset]
+    options = params.permit(*allowed_params)
```
Comment (yoavgeva) [inline app/controllers/edge_controller.rb:43]: you sure you need slice? it's just do a duplicate of list

File: app/controllers/edge_controller.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      variableToReturn = {}
+      variableToReturn[:id] = variable[:resource_id]
+      variableToReturn[:owner] = variable[:owner_id]
+      variableToReturn[:permissions] =  variable.permissions.select{|h| h[:privilege].eql?('execute')}
+      unless variable.last_secret.nil?
+        variableToReturn[:version] = variable.last_secret.version
+        variableToReturn[:value] = variable.last_secret.value
+      end
+      results  << variableToReturn
+    end
```
Comment (yoavgeva) [inline app/controllers/edge_controller.rb:37]: need to be in it's own method

File: app/controllers/edge_controller.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    )
+    results = []
+    hosts = scope.eager(:credentials).all
+    hosts.each do |host|
+      hostToReturn = {}
+      hostToReturn[:id] = host[:role_id]
+      hostToReturn[:api_key] =host.api_key
+      hostToReturn[:memberships] =host.all_roles.all.select{|h| h[:role_id] != (host[:role_id])}
+      results  << hostToReturn
+    end
```
Comment (yoavgeva) [inline app/controllers/edge_controller.rb:61]: need to be in it's own method

File: app/controllers/edge_controller.rb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  end
+
+  def all_hosts
+    allowed_params = %i[account limit offset]
+    options = params.permit(*allowed_params)
+                    .slice(*allowed_params).to_h.symbolize_keys
+    verify_edge_host(options)
+    offset = options[:offset]
+    limit = options[:limit]
+    scope = Role.where(:role_id.like("conjur:host:data%"))
```
Comment (yoavgeva) [inline app/controllers/edge_controller.rb:48]: Is there no Exceptions that can be thrown? 

For example from above I saw 

`rescue ApplicationController::Forbidden
      raise
    rescue ArgumentError => e
      raise ApplicationController::UnprocessableEntity, e.message`

File: cucumber/api/features/step_definitions/user_steps.rb [hunk 6]
```diff
@@ -35,7 +35,7 @@
 
 Given("I login as {string}") do |login|
   if host?(login)
-    loginid = login.split('/')[1]
+    loginid = login.slice(login.index('/')+1, login.length)
```
Comment (yoavgeva) [inline cucumber/api/features/step_definitions/user_steps.rb:38]: I saw you fixed it @amosmintzcyberark, it does not break other tests?

**Review by aloncarmel111 — APPROVED** [review]

**Review by aloncarmel111 — CHANGES_REQUESTED** [review]
.

**Pull request merged**